app/models/miq_server.rb

```diff
@@ -51,18 +51,15 @@ def self.active_miq_servers
   end
 
   def self.atStartup
-    configuration  = VMDB::Config.new("vmdb")
-    starting_roles = configuration.config.fetch_path(:server, :role)
+    starting_roles = ::Settings.server.role
 
     # Change the database role to database_operations
-    roles = configuration.config.fetch_path(:server, :role)
+    roles = starting_roles.dup
     if roles.gsub!(/\bdatabase\b/, 'database_operations')
-      configuration.config.store_path(:server, :role, roles)
+      ::Settings.save!(MiqServer.my_server, :server => {:role => roles})
+      ::Settings.reload!
```

Comment:
This feels wrong.  I think you should be using MiqServer.set_config instead, but perhaps you can't at boot time?